daemon/metrics.go

```diff
@@ -0,0 +1,33 @@
+package daemon
+
+import "github.com/docker/go-metrics"
+
+var (
+	containerActions   metrics.LabeledTimer
+	imageActions       metrics.LabeledTimer
+	daemonActions      metrics.LabeledTimer
+	networkActions     metrics.LabeledTimer
+	engineVersion      metrics.LabeledGauge
+	engineCpus         metrics.Gauge
+	engineMemory       metrics.Gauge
+	healthCheckCounter metrics.LabeledCounter
+)
+
+func init() {
+	ns := metrics.NewNamespace("engine", "daemon", nil)
+	containerActions = ns.NewLabeledTimer("container_actions", "The number of seconds it takes to process each container action", "action")
+	daemonActions = ns.NewLabeledTimer("daemon_actions", "The number of seconds it takes to process each daemon action", "action")
+	networkActions = ns.NewLabeledTimer("network_actions", "The number of seconds it takes to process each network action", "action")
+	engineVersion = ns.NewLabeledGauge("engine", "The version and commit information for the engine process", metrics.Unit("info"),
+		"version",
+		"commit",
+		"architecture",
+		"graph_driver", "kernel",
+		"os",
+	)
+	engineCpus = ns.NewGauge("engine_cpus", "The number of cpus that the host system of the engine has", metrics.Total)
+	engineMemory = ns.NewGauge("engine_memory", "The number of bytes of memory that the host system of the engine has", metrics.Bytes)
+	healthCheckCounter = ns.NewLabeledCounter("health_checks", "The total number of health checks executed", "action")
```

Comment:
Action doesn't make sense as a label here as it's an output rather than an input. As noted below this should be health_checks and health_checks_failed.